server/neptune/workflows/internal/deploy/revision/queue/worker.go

```diff
@@ -1,18 +1,49 @@
 package queue
 
 import (
+	"context"
 	"fmt"
 
 	internalContext "github.com/runatlantis/atlantis/server/neptune/context"
-	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/deployment"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
 	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/deploy/terraform"
 	"go.temporal.io/sdk/temporal"
 	"go.temporal.io/sdk/workflow"
 )
 
-type revisionProcessor interface {
-	Process(ctx workflow.Context, requestedDeployment terraform.DeploymentInfo, latestDeployment *deployment.Info) (*deployment.Info, error)
+type LockStatus string
+
+const (
+	UnlockSignalName = "unlock"
+
+	Unlocked = LockStatus("unlocked")
+	Locked   = LockStatus("locked")
+
+	LockedRootChecksSummary = "The current root has been locked. This is likely the result of a manual deployment. To override that lock and allow the main branch to perform new deployments, select the Unlock button."
+)
+
+func NewLockState() *LockState {
+	return &LockState{
+		status: Unlocked,
+	}
+}
+
+type LockState struct {
```

Comment:
Thoughts on including `Lock` in the queue itself and exposing these methods on the lock: 

- `Lock()`
-  `Unlock()`
-  `GetLockStatus()`

Then, we wouldn't have to pass queue and lock state separately unless we plan to use `LockState` on it's own!